bin/c8.js

```diff
@@ -8,31 +8,48 @@ const waitTillPortOpen = require('wait-till-port-open')
 
 getPort().then(async port => {
   foreground(
-    ['node', `--inspect-brk=${port}`].concat(process.argv.slice(2)),
+    [process.execPath, `--inspect-brk=${port}`].concat(process.argv.slice(2)),
     (done) => {
       console.info('actually got here')
+      done()
     }
   )
   try {
     await waitTillPortOpen(port)
     const client = await CRI({port: port})
 
+    const initialPause = new Promise((resolve) => {
+      client.once('Debugger.paused', resolve)
+    })
+
+    const mainContextInfo = new Promise((resolve) => {
+      client.once('Runtime.executionContextCreated', (message) => {
+        resolve(message.context)
+      })
+    })
+
+    const executionComplete = new Promise((resolve) => {
+      client.on('Runtime.executionContextDestroyed', async (message) => {
+        if (message.executionContextId === (await mainContextInfo).id) {
+          resolve(message)
+        }
+      })
+    })
+
     const {Debugger, Runtime, Profiler} = client
-    await Runtime.runIfWaitingForDebugger()
-    await Runtime.enable()
-    await Profiler.enable()
+    await Promise.all([
+      Profiler.enable(),
+      Runtime.enable(),
+      Debugger.enable(),
+      Runtime.runIfWaitingForDebugger(),
+      initialPause
```

Comment:
weirdly, this does not seem to work for `process.exit` or `throw` in a script, unless you refactor thusly:

```js
    const {Debugger, Runtime, Profiler} = client
    await Promise.all([
      Profiler.enable(),
      Runtime.enable(),
      Debugger.enable(),
      Runtime.runIfWaitingForDebugger(),
    ])
    await Profiler.startPreciseCoverage({callCount: true, detailed: true})
    await initialPause
    await Debugger.resume()
```

you can exercise this by running `./bin/c8.js test/fixtures/c.mjs` (which should output block coverage information when running properly) -- any idea why that order of operations matters? feels a little nit-picky.

Reply:
The proposed solution crashes with a segfault here. Will investigate further.

Reply:
Some more discoveries:

1. The reason why nothing in `c.mjs`, the main file, gets tracked is that `Profiler.startPreciseCoverage` was executed *after* the main file has been parsed. As such, you are correct in discovering that `Profiler.startPreciseCoverage` needs to be executed as early as possible.
2. However, the V8 used in Node.js 9.x and master *crashes* with a segmentation fault if `Profiler.startPreciseCoverage` is received before `Debugger.paused` -- but not when `Profiler.startPreciseCoverage` is received, rather it continues normally until when `Debugger.paused` is expected to be emitted.
3. The V8 used in canary builds (6.4.358-node.0 tested) does *not* exhibit the crash above, and works fully as expected, due to https://github.com/v8/v8/commit/1420e44db0ac3631687deb9fc6816ac97b9f499c. I will look into backporting this to Node.js.